scripts/login_signup.js

```diff
@@ -0,0 +1,134 @@
+import { isValidEmail, isValidPassword } from "./utils/validate.js";
+
+const form = document.forms[0];
+const email = form["email"];
+const nickname = form["nickname"];
+const password = form["password"];
+const passwordConfirm = form["passwordConfirm"];
+const submit = form.lastElementChild;
+
+// 폼 유효성 & 활성화 초기화
+
+let isEmailValid = false;
+let isNicknameValid = false;
+let isPasswordValid = false;
+let isPasswordConfirmValid = false;
+submit.disabled = true;
+
+// 폼 제출 활성화 판단
+
+function updateSubmit() {
+  let isFormValid = isEmailValid && isPasswordValid;
+
+  if (!!nickname)
+    isFormValid = isFormValid && isNicknameValid && isPasswordConfirmValid;
+
+  submit.disabled = !isFormValid;
+}
+
+// 폼 제출 이벤트 변경
+
+form.addEventListener("submit", (e) => {
+  e.preventDefault();
+  if (!!nickname) document.location.href = "/signin.html";
+  else document.location.href = "/items.html";
+});
```

Comment:
P2:
두번 변환하는 것보다 한번만 변환하는 것이 이해하기 쉽습니다. 아래와 같은 방식을 추천드립니다.

```suggestion
form.addEventListener("submit", (e) => {
  e.preventDefault();
  if (!nickname) return document.location.href = "/items.html";
  document.location.href = "/signin.html";
});
```

Reply:
조건식에 값이 담길 때,  boolean 용도임을 명시하기 위해 `!!`를 적용했는데 불필요하다면 수정하겠습니다!

Reply:
앗 `!`를 한번만 써도 boolean 값으로 변환이 되기 때문에 `!nickname => 없을 시 true, 있을 시 false`, `!!nickname => 없을 시 false, 있을 시 true` 중에 한번만 변환하는게 더 이해하기 쉬울거라고 생각했습니다.